cmd/manager/main.go

```diff
@@ -31,9 +32,11 @@ var (
 
 	debug = app.Flag("debug", "Run with debug logging").Short('d').Bool()
 
-	cloudProviderName = app.Flag("cloud-provider", "Which cloud provider to use, options: [aws]").Default("aws").String()
-	addr              = app.Flag("address", "Address to listen on for /metrics").Default(":8080").String()
-	namespace         = app.Flag("namespace", "Namespace to watch for cycle request objects").Default("kube-system").String()
+	cloudProviderName     = app.Flag("cloud-provider", "Which cloud provider to use, options: [aws]").Default("aws").String()
+	messagingProviderName = app.Flag("messaging-provider", "Which message provider to use, options: [slack]").Default("slack").String()
```

Comment:
I feel like it would more clear for default to be `""` (empty/none)?
Then below don't build the provider if it's not set. Hiding the "not enabled" logic in the `BuildNotifier` return is a little obfuscated (nil, nil). Thoughts @awprice @patrickshan ?
```
if *messagingProviderName != "" {
    notifier, err := notifierbuilder.BuildNotifier(*messagingProviderName)
    if err != nil {
        log.Error(err, "Unable to build notifier")
	os.Exit(1)
    }
}
```

e.g. I had to dig into the Slack Builder code to see that 
```
// Return nil without an error if both tokens are missing since this is an optional feature
	if token == "" {
		return nil, nil
	}
```
Handling the on/off logic in the main will also be applicable across all providers, not just slack with a `TOKEN`

Also would make the help message make more sense if messaging is optional
```diff
-     --messaging-provider="slack"
+     --messaging-provider=""  
                                 Which message provider to use, options: [slack]
```

Reply:
+1 for make it default to "" which makes it clearer for switching on/off notification feature.

Reply:
Agree on your points, handling the logic in the main is clearer. Also agree on making the default "".